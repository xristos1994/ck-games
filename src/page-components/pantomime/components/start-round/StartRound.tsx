import React, { FC, ReactElement } from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { classnames } from "@utils/component-utils";
import { Button } from "@components";
import { startRound, goBack, setMovie } from "@models/pantomime/actions";
import {
  teamNameThatPlaysNow,
  canGoBack,
  availableMovies,
  movie,
} from "@models/pantomime/props";
import { IState } from "@models/interfaces";

const styles = require("./styles.module.css");

interface IProps {
  canGoBack: boolean;
  teamNameThatPlaysNow: string;
  availableMovies: [string, string];
  movie: string;
  startRound: () => void;
  goBack: () => void;
  setMovie: (movie: string) => void;
}

const _StartRound: FC<IProps> = ({
  startRound,
  teamNameThatPlaysNow,
  availableMovies,
  movie,
  canGoBack,
  goBack,
  setMovie,
}): ReactElement => {
  return (
    <div className={styles.startRoundContainer}>
      <div className={classnames(styles.playsNow)}>
        {teamNameThatPlaysNow} plays now
      </div>
      {availableMovies.map(_movie => (
        <Button
          key={_movie}
          other={{ disabled: _movie === movie }}
          onClick={() => setMovie(_movie)}
          className={classnames(styles.movieButton, {
            [styles.selectedMovieButton]: _movie === movie,
            [styles.notSelectedMovieButton]: _movie !== movie,
          })}
        >
          {_movie}
        </Button>
      ))}
      <Button
        other={{ disabled: availableMovies.indexOf(movie) === -1 }}
        onClick={() => startRound()}
        className={classnames(styles.startRoundButton)}
      >
        PROCEED
      </Button>
      {canGoBack && (
        <Button
          onClick={() => goBack()}
          className={classnames(styles.backButton)}
        >
          BACK
        </Button>
      )}
    </div>
  );
};

const StartRound = connect(
  createStructuredSelector<
    IState,
    {
      canGoBack: IProps["canGoBack"];
      teamNameThatPlaysNow: IProps["teamNameThatPlaysNow"];
      availableMovies: IProps["availableMovies"];
      movie: IProps["movie"];
    },
    {
      canGoBack: IProps["canGoBack"];
      teamNameThatPlaysNow: IProps["teamNameThatPlaysNow"];
      availableMovies: IProps["availableMovies"];
      movie: IProps["movie"];
    }
  >({
    teamNameThatPlaysNow,
    canGoBack,
    availableMovies,
    movie,
  }),
  { startRound, goBack, setMovie }
)(_StartRound);

export { StartRound };
