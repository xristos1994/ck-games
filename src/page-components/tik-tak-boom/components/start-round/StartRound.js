import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { classnames } from "@utils/component-utils";
import { Button } from "@components";
import styles from "./styles.module.css";
import { startRound } from "@models/tik-tak-boom/actions";
import { playerNameThatStartsRound } from "@models/tik-tak-boom/props";
import { ScoreBoard } from "./../";

const _StartRound = ({ startRound, playerNameThatStartsRound }) => {
  return (
    <div className={styles.startRoundContainer}>
      <div
        className={classnames(
          styles.playsFirst,
          "extraLargeText",
          "main-color"
        )}
      >
        {playerNameThatStartsRound} plays first
      </div>
      <ScoreBoard />
      <Button
        onClick={() => startRound()}
        className={classnames(
          styles.startRoundButton,
          "extraLargeText",
          "primary-dark"
        )}
      >
        PROCEED
      </Button>
    </div>
  );
};

const StartRound = connect(
  createStructuredSelector({
    playerNameThatStartsRound,
  }),
  { startRound }
)(_StartRound);

export { StartRound };
