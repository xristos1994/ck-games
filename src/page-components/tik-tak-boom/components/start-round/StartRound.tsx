import React, { FC, ReactElement } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { classnames, compose } from '@utils/component-utils';
import { withTranslation, ITranslate } from '@models/i18n/hoc';
import { Button } from '@components';
import { startRound, goBack } from '@models/tik-tak-boom/actions';
import { playerNameThatStartsRound, canGoBack } from '@models/tik-tak-boom/props';
import { IState } from '@models/interfaces';
import { ScoreBoard } from './../score-board'; // Alias "page-components/tik-tak-boom/components";
import styles from './styles.module.css';

interface IProps {
  canGoBack: boolean;
  playerNameThatStartsRound: string | null;
  startRound: () => void;
  goBack: () => void;
  t: ITranslate;
}

const _StartRound: FC<IProps> = ({ startRound, playerNameThatStartsRound, canGoBack, goBack, t }): ReactElement => {
  return (
    <div className={styles.startRoundContainer}>
      <div className={classnames(styles.playsFirst)}>
        {t('Player, it is your turn', [playerNameThatStartsRound as string])}
      </div>
      <ScoreBoard />
      <Button onClick={() => startRound()} className={classnames(styles.startRoundButton)}>
        {t('CONTINUE')}
      </Button>
      {canGoBack && (
        <Button onClick={() => goBack()} className={classnames(styles.backButton)}>
          {t('BACK')}
        </Button>
      )}
    </div>
  );
};

const StartRound = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        canGoBack: IProps['canGoBack'];
        playerNameThatStartsRound: IProps['playerNameThatStartsRound'];
      },
      {
        canGoBack: IProps['canGoBack'];
        playerNameThatStartsRound: IProps['playerNameThatStartsRound'];
      }
    >({
      playerNameThatStartsRound,
      canGoBack
    }),
    { startRound, goBack }
  ),
  withTranslation
)(_StartRound);

export { StartRound };
