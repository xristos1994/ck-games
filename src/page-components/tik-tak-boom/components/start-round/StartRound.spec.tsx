import React, { MouseEventHandler, ReactChild } from 'react';
import renderer from 'react-test-renderer';
import { _StartRound as StartRound } from './StartRound';

const ScoreBoard = () => {
  return (
    <div>
      ScoreBoard
    </div>
  );
};

const Button = (
  { children, onClick, className='' } :
  { children: ReactChild | ReactChild[]; onClick: MouseEventHandler<HTMLButtonElement>; className: string }
) => {
  return (
    <button onClick={onClick} className={className || undefined}>
      {children}
    </button>
  );
};

jest.mock('./../score-board', () => ({
  __esModule: true,
  ScoreBoard: ScoreBoard
}));

jest.mock('@components/', () => ({
  __esModule: true,
  Button: Button
}));

const startRound = () => void 0;
const playerNameThatStartsRound = 'Player X';
const goBack = () => void 0;
const t = (label: string, param?: string[]) => label + (param ? param[0] : '');

describe('page-components/TikTakBoom/StartRound', () => {
  it('renders correctly', () => {
    let tree = renderer.create(
      <StartRound
        startRound={startRound}
        playerNameThatStartsRound={playerNameThatStartsRound}
        canGoBack={true}
        goBack={goBack}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <StartRound
        startRound={startRound}
        playerNameThatStartsRound={playerNameThatStartsRound}
        canGoBack={false}
        goBack={goBack}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
