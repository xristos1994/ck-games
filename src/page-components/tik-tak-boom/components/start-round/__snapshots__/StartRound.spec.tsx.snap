// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/TikTakBoom/StartRound renders correctly 1`] = `
<div
  className="startRoundContainer"
>
  <div
    className="playsFirst"
  >
    Player, it is your turnPlayer X
  </div>
  <div>
    ScoreBoard
  </div>
  <button
    className="startRoundButton"
    onClick={[MockFunction]}
  >
    CONTINUE
  </button>
  <button
    className="backButton"
    onClick={[MockFunction]}
  >
    BACK
  </button>
</div>
`;

exports[`page-components/TikTakBoom/StartRound renders correctly 2`] = `
<div
  className="startRoundContainer"
>
  <div
    className="playsFirst"
  >
    Player, it is your turnPlayer X
  </div>
  <div>
    ScoreBoard
  </div>
  <button
    className="startRoundButton"
    onClick={[MockFunction]}
  >
    CONTINUE
  </button>
</div>
`;
